src/domain/tokens/useInfoTokens.ts

```diff
@@ -0,0 +1,219 @@
+import { getTokens, getWhitelistedTokens } from "./utils";
+import { getContract } from "../../config/contracts";
+import useSWR from "swr";
+import { contractFetcher } from "../../lib/contracts";
+import VaultReader from "../../abis/VaultReader.json";
+import {
+  BASIS_POINTS_DIVISOR,
+  DEFAULT_MAX_USDG_AMOUNT,
+  MAX_PRICE_DEVIATION_BASIS_POINTS,
+  USD_DECIMALS,
+  USDG_ADDRESS,
+} from "../../lib/legacy";
+import { getServerUrl } from "../../config/backend";
+import { Token, TokenInfo } from "./types";
+import { BigNumber } from "ethers";
+import { Web3ReactContextInterface } from "@web3-react/core/dist/types";
+import { bigNumberify, expandDecimals } from "../../lib/numbers";
+
+export function useInfoTokens(
+  library: Web3ReactContextInterface["library"],
+  chainId: number,
+  active: boolean,
+  tokenBalances: BigNumber[] | undefined,
```

Comment:
oh it's not the same is `tokenBalances?: BigNumber[]` ?

Reply:
Hm, in this place it's true, fixed it

but in general it is not quite the same thing: if you define a parameter as `tokenBalances?`
 it will be considered optional, and in that case you cannot specify the following parameters as mandatory as in the screenshot.

tokenBalances: BigNumber[] | undefined allows to explicitly specify undefined instead of parameter and types are not broken 
 
 <img width="603" alt="Screenshot 2022-09-29 at 22 13 29" src="https://user-images.githubusercontent.com/110596516/193111250-0f2ce207-e71d-4df3-9ca4-8552c8982459.png">